frontend/src/metabase/setup/reducers.ts

```diff
@@ -54,6 +60,12 @@ export const reducer = createReducer(initialState, builder => {
     state.usageReason = usageReason;
     state.step = getNextStep({ setup: state } as State);
   });
+  builder.addCase(submitLicenseToken.pending, (state, { meta }) => {
+    const token = meta.arg;
+    state.licenseToken = token;
+    state.step = getNextStep({ setup: state } as State);
```

Comment:
I was confused again on why we don't have to call any navigation function inside the Token step. I'd be great to make sure we don't do the same thing in different places.

Is it a plan to move all the navigation logic to the reducer as well or do we need to still have navigation logics in both React components and in reducer?

Reply:
I was confused this with `handleStepSelect`, so all good here.